purge_cluster.py

```diff
@@ -0,0 +1,79 @@
+"""Purge Ceph Cluster.
+
+Usage:
+    purge_cluster.py [--installer <IP>] [--username <USERNAME>] [--password <PASSWORD>]
+
+Options:
+    -h --help       Show this help message and exit.
+    --installer <IP>     Installer node IP.
+    --username <USERNAME>   SSH username.
+    --password <PASSWORD>   SSH password.
+
+Assumptions :
+    This will assume all the ceph nodes will have same root username password as provided for installer node
+"""
+
+import json
+
+import paramiko
+from docopt import docopt
+
+
+def execute_ssh_command(ssh_client, command):
+    stdin, stdout, stderr = ssh_client.exec_command(command)
+    return stdout.read().decode("utf-8")
+
+
+def get_fsid(ssh_client):
+    fsid_command = "cephadm shell -- ceph fsid -f json"
+    fsid_output = execute_ssh_command(ssh_client, fsid_command)
+    fsid_json = json.loads(fsid_output)
+    return fsid_json["fsid"]
+
+
+def get_node_ips(ssh_client):
+    node_ips_command = "cephadm shell -- ceph orch host ls -f json"
+    node_ips_output = execute_ssh_command(ssh_client, node_ips_command)
+    node_ips_json = json.loads(node_ips_output)
+    return [node["addr"] for node in node_ips_json]
+
+
+def main():
+    args = docopt(__doc__)
+
+    installer_node_ip = args["--installer"]
+    username = args["--username"]
+    password = args["--password"]
+
+    ssh_installer = paramiko.SSHClient()
+    ssh_installer.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh_installer.connect(installer_node_ip, username=username, password=password)
+
+    fsid = get_fsid(ssh_installer)
+
+    node_ips = get_node_ips(ssh_installer)
+    # print("Node IPs:", node_ips)
+    node_ips.remove(installer_node_ip)
+    print("Cluster Node IPs list:", node_ips)
+    for node_ip in node_ips:
```

Comment:
Add a check to make sure that cephadm package is installed on all the nodes.

This would be important if we want to use this standalone script to purge manually deployed clusters.

Reply:
If the cephadm is absent in any of the nodes, the process will encounter an error.
Here are options to address this situation:
Users have the option to manually install cephadm on the respective node and subsequently relaunch the script.
As a preemptive measure, we execute the cephadm rm-cluster command on the installer node as the final step after it has already been run on the other nodes.

As part of the cephci procedure, it's ensured that cephadm is installed across all nodes.

Reply:
Yes correct, it would error out. What I am suggesting to add a check so that it does not fail upon command execution in the middle of script execution( this would leave with only few nodes with no cephadm daemons ) , but we do a pre check and inform user about the same.

Reply:
cephadm rm-cluster was removing the cluster logs as well

```
[root@mero014 ~]# ls -lrt /var/log/ceph/
total 552
-rw-r--r--. 1 root root  30005 Aug  3 23:59 cephadm.log-20230804.gz
-rw-r--r--. 1 root root  30062 Aug  4 23:59 cephadm.log-20230805.gz
-rw-r--r--. 1 root root  30104 Aug  5 23:59 cephadm.log-20230806.gz
-rw-r--r--. 1 root root  30064 Aug  6 23:59 cephadm.log-20230807.gz
-rw-r--r--. 1 root root  45867 Aug  7 23:59 cephadm.log-20230808.gz
-rw-r--r--. 1 root root  40914 Aug  8 23:59 cephadm.log-20230809.gz
-rw-r--r--. 1 root root  29785 Aug  9 23:59 cephadm.log-20230810.gz
drwxrwx---. 2  167  167   4096 Aug 10 00:00 2ecf8754-3614-11ee-a05b-ac1f6b40d304
-rw-r--r--. 1 root root 302084 Aug 10 19:14 cephadm.log
[root@mero014 ~]# 
[root@mero014 ~]# 
[root@mero014 ~]# ls -lrt /var/log/ceph/
total 604
-rw-r--r--. 1 root root  30005 Aug  3 23:59 cephadm.log-20230804.gz
-rw-r--r--. 1 root root  30062 Aug  4 23:59 cephadm.log-20230805.gz
-rw-r--r--. 1 root root  30104 Aug  5 23:59 cephadm.log-20230806.gz
-rw-r--r--. 1 root root  30064 Aug  6 23:59 cephadm.log-20230807.gz
-rw-r--r--. 1 root root  45867 Aug  7 23:59 cephadm.log-20230808.gz
-rw-r--r--. 1 root root  40914 Aug  8 23:59 cephadm.log-20230809.gz
-rw-r--r--. 1 root root  29785 Aug  9 23:59 cephadm.log-20230810.gz
-rw-r--r--. 1 root root 359469 Aug 10 19:14 cephadm.log
```
For safe side, i have incorporated rm -rf /var/log/ceph/*

Reply:
Yes. `cephadm rm-cluster` will remove ceph logs, but the cephadm logs would still be present. 

Yes, it's good to have  `rm -rf /var/log/ceph/*`